wpilibc/src/main/native/include/frc/ResistanceCalculator.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <wpi/circular_buffer.h>
+
+namespace frc {
+
+/**
+ * Finds the resistance of a channel or the entire robot using a running linear
+ * regression over a window. Must be updated with current and voltage
+ * periodically using the <code>Calculate</code> method. <p>To use this for
+ * finding the resistance of a channel, use the calculate method with the
+ * battery voltage minus the voltage at the motor controller or whatever is
+ * plugged in to the PDP at that channel.</p>
+ */
+class ResistanceCalculator {
+ public:
+  static constexpr int kDefaultBufferSize = 250;
+  static constexpr double kDefaultRSquaredThreshold = 0.5;
+
+  /**
+   * Create a ResistanceCalculator to find the resistance of a channel using a
+   * running linear regression over a window. Must be updated with current and
+   * voltage periodically using the <code>Calculate</code> method.
+   *
+   * @param bufferSize The maximum number of points to take the linear
+   * regression over.
+   * @param krSquaredThreshold The minimum R^2 value considered significant
+   * enough to return the regression slope instead of NaN.
```

Comment:
Use `R²`. Also, the valid range for it should be specified, and what the different ends of the range mean.